pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -234,8 +236,12 @@ func CreateGCECloud(projectID, region, zone string, managedZones []string, netwo
 	}, nil
 }
 
-// Initialize passes a Kubernetes clientBuilder interface to the cloud provider
-func (gce *GCECloud) Initialize(clientBuilder controller.ControllerClientBuilder) {}
+// Initialize takes in a clientBuilder and spawns a goroutine for watching the clusterid configmap.
+// This must be called before utilizing the funcs of gce.ClusterId
+func (gce *GCECloud) Initialize(clientBuilder controller.ControllerClientBuilder) {
```

Comment:
Should this block waiting for the initialization? Is the rest of the code ok with this as an async update?